# DESAFIOS SEMANA 05

## Desafio 1 - Ler arquivos

In [20]:
import pandas as pd

df_csv = pd.read_csv("dataset/cadastro_funcionario.csv")
df_excel = pd.read_excel("dataset/recursos_humanos.xlsx")
df_json = pd.read_json("dataset/emails_corporativos.json")


# df.head()
print("df.head()") 
print("Mostrar as 5 primeiras linhas do DataFrame")
print("Também é possível escolher a quantidade entre parentes()")
display(df_csv.head())
display(df_csv.head(2))


# df.info()
print("\n")
print("df.info()")
print("Ele responde:\n",
      "Quantas linhas existem?\n",
      "Quantas colunas?\n",
      "Qual o tipo de cada coluna?\n",
      "Existem valores vazios?\n")
display(df_excel.info())


# df.describe()
print()
print("df.describe()")
print("Esse comando faz calculos de estatísticas automáticas nas colunas numéricas.")
print("count = Quantidade de valores.")
print("mean = Média.")
print("std = Desvio padrão. Quanto os dados variam.")
print("min = Menor valor.")
print("max = Maior valor.")
print("50% = Mediana.")
print("25% = Primeiro quartil.")
print("75% = Terceiro quartil.")
display(df_json.describe())


# df.shape
print()
print("df.shape")
print("Mostra a quantidade de (linhas, colunas)")
print(df_csv.shape)

print("Pode pegar separadamente.")
print("df_csv.shape[0] = linhas")
print(df_csv.shape[0])

print("df_csv.shape[1] = colunas")
display(df_csv.shape[1])


# df.columns
print()
print("df.columns")
print("Mostra apenas os nomes das colunas.")
print(df_json.columns)

df.head()
Mostrar as 5 primeiras linhas do DataFrame
Também é possível escolher a quantidade entre parentes()


,id_funcionario,funcionario,cargo,setor
0,1,Ana Souza,Atendente,Comercial
1,2,Bruno Lima,Motorista,Logística
2,3,Carlos Mendes,Separador de Estoque,Estoque
3,4,Daniela Rocha,Atendente,Comercial
4,5,Eduardo Santos,Motorista,Logística


,id_funcionario,funcionario,cargo,setor
0,1,Ana Souza,Atendente,Comercial
1,2,Bruno Lima,Motorista,Logística




df.info()
Ele responde:
 Quantas linhas existem?
 Quantas colunas?
 Qual o tipo de cada coluna?
 Existem valores vazios?

<class 'pandas.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   id_funcionario  30 non-null     int64
 1   funcionario     30 non-null     str  
 2   cargo           30 non-null     str  
 3   setor           30 non-null     str  
dtypes: int64(1), str(3)
memory usage: 1.1 KB


None


df.describe()
Esse comando faz calculos de estatísticas automáticas nas colunas numéricas.
count = Quantidade de valores.
mean = Média.
std = Desvio padrão. Quanto os dados variam.
min = Menor valor.
max = Maior valor.
50% = Mediana.
25% = Primeiro quartil.
75% = Terceiro quartil.


,id_funcionario
count,30.000000
mean,15.500000
std,8.803408
min,1.000000
25%,8.250000
50%,15.500000
75%,22.750000
max,30.000000



df.shape
Mostra a quantidade de (linhas, colunas)
(30, 4)
Pode pegar separadamente.
df_csv.shape[0] = linhas
30
df_csv.shape[1] = colunas


4


df.columns
Mostra apenas os nomes das colunas.
Index(['id_funcionario', 'funcionario', 'email'], dtype='str')


## Desafio 2 - Selecionar dados

In [61]:
import pandas as pd

df = pd.read_csv("dataset/cadastro_funcionario.csv")

# print("Mostrar os nomes das colunas:")
# display(df.columns) 


# print("\n")
# print("Mostrar apenas os resultados da coluna 'funcionario':")
# print(df["funcionario"]) 


# print("\n")
# print("Mostrar os resultadas das colunas 'funcionario' e 'cargo':")
# print(df[["funcionario","cargo"]])

# display(df)
# print("\n")
# print("Mostrar o resultado pela posição (índice numérico)")
# display(df.iloc[0])

# print("\n")
# print("Mostrar o resultado pela posição (índice + nome do índice)")
# display(df.loc[0,"cargo"])


# print("\n")
# print("Filtros")
display(df[df["cargo"]=="Atendente"])
display(df[df["setor"]=="Estoque"])

,id_funcionario,funcionario,cargo,setor
0,1,Ana Souza,Atendente,Comercial
3,4,Daniela Rocha,Atendente,Comercial
9,10,Juliana Neves,Atendente,Comercial
13,14,Olivia Ferreira,Atendente,Comercial
17,18,Sabrina Moura,Atendente,Comercial
22,23,Ximena Lopes,Atendente,Comercial
26,27,Beatriz Cunha,Atendente,Comercial


,id_funcionario,funcionario,cargo,setor
2,3,Carlos Mendes,Separador de Estoque,Estoque
6,7,Gabriel Oliveira,Separador de Estoque,Estoque
10,11,Lucas Alves,Separador de Estoque,Estoque
11,12,Mariana Castro,Supervisora,Estoque
14,15,Paulo Gomes,Separador de Estoque,Estoque
18,19,Thiago Cardoso,Separador de Estoque,Estoque
20,21,Vinicius Barbosa,Separador de Estoque,Estoque
23,24,Yago Nascimento,Auxiliar de Estoque,Estoque
27,28,Caio Freitas,Separador de Estoque,Estoque


## Desafio 3 - Criar colunas

In [ ]:
df["Total"]
df["Bonus"]
df["Lucro"]

# Total = Quantidade * Valor

## Desafio 4 - Estatísticas

In [ ]:
sum()
mean()
max()
min()
count()
value_counts()

## Desafio 5 - Agrupamentos.

In [ ]:
groupby()
# Vendedor
# Total vendido
# Cidade
# Quantidade

## Desafio 6 - Ordenação

In [ ]:
sort_values()

# Top 5 vendedores
# Top 10 produtos
# Maior salário
# Menor salário

## Merge e Pivot
## Desafio 7 - Juntar duas tabelas.

In [ ]:
merge()

## Desafio 8 - Pivot Table
Transformar: Categoria >> Região

Em uma tabela.

## Desafio 9 - NumPy

In [ ]:
Criar Arrays
array()
zeros()
ones()
arange()

## Desafio 10 - Operações
Sem utilizar for e while

In [ ]:
+
-
*
/
**

## Desafio 11 - Vetorização
O exercício inteiro será: preco * quantidade

e depois: preco * 1.10

Entender Broadcasting.

# Desafio 12A - Uma função

In [ ]:
def calcular_total(preco, quantidade):
    return preco * quantidade

## Desafio 12B - Duas funções.

In [ ]:
def calcular_total()
def calcular_desconto()

total = calcular_total()
final = calcular_desconto(total)

## Desafio 12C - Três funções

In [ ]:
def calcular_total()
def calcular_bonus()
def classificar()